### PR TITLE
Convert approx_distinct to UDAF

### LINKED ISSUE #10837: Convert `approx_distinct` to UDAF
### Is your feature request related to a problem or challenge?

Similar to #10713 #10836 #10834 and others in #8708 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Lordworms**
## Which issue does this PR close?

Closes #10837

### CODE REVIEW

File: datafusion/functions-aggregate/src/approx_distinct.rs [hunk 1]
```diff
@@ -304,3 +223,88 @@ where
...
+
+impl Default for ApproxDistinct {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub struct ApproxDistinct {
+    signature: Signature,
+    alias: Vec<String>,
```
Comment (goldmedal) [inline datafusion/functions-aggregate/src/approx_distinct.rs:244]: ApproxDistinct has no any alias. I think we can just remove this field and use default `AggregateUDFImpl::alias`

File: datafusion/physical-expr/src/aggregate/hyperloglog.rs [hunk 2]
```diff
@@ -115,6 +115,7 @@ where
 
     /// Get the register histogram (each value in register index into
     /// the histogram; u32 is enough because we only have 2**14=16384 registers
+    #[allow(dead_code)]
```
Comment (goldmedal) [inline datafusion/physical-expr/src/aggregate/hyperloglog.rs:118]: After moving to `function-aggregate`, I think we can remove the original file `datafusion/physical-expr/src/aggregate/hyperloglog.rs`.

File: datafusion/physical-expr/src/expressions/mod.rs [hunk 3]
```diff
@@ -64,6 +63,7 @@ pub use crate::window::ntile::Ntile;
 pub use crate::window::rank::{dense_rank, percent_rank, rank, Rank, RankType};
 pub use crate::window::row_number::RowNumber;
 pub use crate::PhysicalSortExpr;
+pub use datafusion_functions_aggregate::approx_distinct::ApproxDistinct;
```
Comment (goldmedal) [inline datafusion/physical-expr/src/expressions/mod.rs:66]: I guess we don't need to public use it.

**Review by goldmedal — COMMENTED** [review]
Thanks @Lordworms 
I think overall good but some suggestions.

File: datafusion/functions-aggregate/src/approx_distinct.rs (hunk 1 above)
Comment (Lordworms) [reply datafusion/functions-aggregate/src/approx_distinct.rs:244]: Sure

File: datafusion/physical-expr/src/aggregate/hyperloglog.rs (hunk 2 above)
Comment (Lordworms) [reply datafusion/physical-expr/src/aggregate/hyperloglog.rs:118]: sure

**Comment by Lordworms** [comment]
I have resolved all the reviews, plz merge this first to avoid an extra conflicts solve, Thanks for your review @goldmedal

**Review by jayzhan211 — APPROVED** [review]

**Comment by jayzhan211** [comment]
Thanks @Lordworms and @goldmedal

**Pull request merged**